app/Http/Controllers/TaxonController.php

```diff
@@ -27,11 +38,22 @@ public function index()
      */
     public function show($id)
     {
-        $taxon = \App\Taxon::find($id);
-        if($taxon){
-            $devices = $taxon->devices();
+        $taxon = Taxon::find($id);
+        $compId = Input::get('company');
+        $this->debug_to_console($compId);
+        if(isset($compId)){
+            if($taxon){
+                $this->debug_to_console('ENTRATO');
+                $devices = $taxon->devices()->where('company_id', $compId)->get();
+                return view('taxon.show', ['taxon' => $taxon, 'devices' => $devices]);
+            }
+        }
+        else{
+            if($taxon){
+                $devices = $taxon->devices()->get();
+                return view('taxon.show', ['taxon' => $taxon, 'devices' => $devices]);
```

Comment:
also better to move the return outside of the condition, since it's the same for both cases